src/modules/context/PatientsContextReduxer.js

```diff
@@ -0,0 +1,27 @@
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "SET_PATIENT": {
+      const { patient: setPatient } = action.payload;
+      const { patients } = state;
+
+      const patientIndex = state.patients.findIndex((patient) => {
+        return patient.id === setPatient.id;
+      });
+
+      if (patientIndex !== -1) {
+        patients[patientIndex] = setPatient;
+      } else {
+        patients.append(setPatient);
+      }
+
+      return {
+        ...state,
```

Comment:
czy tutaj nie dałoby się inaczej tego zapisać pod kątem nazewnictwa żebyśmy zwracali np. state.previousPatients? chyba że  tylko dla mnie to dziwnie wygląda :)

Reply:
Niestety, ale nie jest to możliwe. Nawiązując do [dokumentacji](https://pl.reactjs.org/docs/hooks-reference.html#usereducer) funkcji useReducer:
"...Przyjmuje reduktor (ang. reducer), będący funkcją o sygnaturze (stan, akcja) => nowyStan..."
Z tego wynika, że reduktur musi nam zwrócić cały stan, który jest w kontekście, a nie tylko jego część. 
Akurat w naszym przypadku w stanie przechowujemy tylko listę pacjentów i to co napisałas mogłoby nawet zadziałać (musiałoby być prawdoopodnie jeszcze w obiekcie - { patients: state.previousPatients }, ale przy dodaniu czegokoliwek do kontekstu tracilibyśmy te pozostałe informacje.

BTW jeśli chodzi o zapis "...state", to tutaj korzystam z operatora "..." (spread opetator). Jest on bardzo często używany w js [dokumentacja](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Operatory/Spread_operator). W tym przypadku umożliwia on zastąpienie wybranych atrybutów obiektu. Najpierw rozwijamy cały stan, używając spreada (...state) i później zastępujemy tylko te atrybuty, które chcemy podmienić w obiekcie (patients: newPatients)